luacompiler/tests/integration_test.rs

```diff
@@ -0,0 +1,99 @@
+extern crate luacompiler;
+
+use luacompiler::LuaParseTree;
+use luacompiler::bytecode::LuaBytecode;
+use luacompiler::bytecode::instructions::{ make_instr, Opcode };
+
+#[test]
+fn ldi_generation() {
+    let pt = LuaParseTree::new("./tests/lua_files/ldi.lua");
+    let bc = pt.unwrap().compile_to_ir();
+    assert_eq!(bc.instrs_len(), 2);
+    assert_eq!(bc.reg_count(), 2);
+    assert_eq!(bc.get_int(0), 1);
+    assert_eq!(bc.get_instr(0), make_instr(Opcode::LDI, 0, 0, 0));
+    assert_eq!(bc.get_instr(1), make_instr(Opcode::MOV, 1, 0, 0));
+}
+
+#[test]
+fn ldf_generation() {
+    let pt = LuaParseTree::new("./tests/lua_files/ldf.lua");
+    let bc = pt.unwrap().compile_to_ir();
+    assert_eq!(bc.instrs_len(), 2);
+    assert_eq!(bc.reg_count(), 2);
+    assert_eq!(bc.get_float(0).to_string(), "2");
+    assert_eq!(bc.get_instr(0), make_instr(Opcode::LDF, 0, 0, 0));
+    assert_eq!(bc.get_instr(1), make_instr(Opcode::MOV, 1, 0, 0));
+}
+
+#[test]
+fn lds_generation() {
+    let pt = LuaParseTree::new("./tests/lua_files/lds.lua");
+    let bc = pt.unwrap().compile_to_ir();
+    assert_eq!(bc.instrs_len(), 2);
+    assert_eq!(bc.reg_count(), 2);
+    assert_eq!(bc.get_string(0), "1.2");
+    assert_eq!(bc.get_instr(0), make_instr(Opcode::LDS, 0, 0, 0));
+    assert_eq!(bc.get_instr(1), make_instr(Opcode::MOV, 1, 0, 0));
+}
+
+fn assert_bytecode(opcode: Opcode, bc: &LuaBytecode) {
+    assert_eq!(bc.instrs_len(), 4);
+    assert_eq!(bc.reg_count(), 4);
+    assert_eq!(bc.get_int(0), 1);
+    assert_eq!(bc.get_int(1), 2);
+    assert_eq!(bc.get_instr(0), make_instr(Opcode::LDI, 0, 0, 0));
+    assert_eq!(bc.get_instr(1), make_instr(Opcode::LDI, 1, 1, 0));
+    assert_eq!(bc.get_instr(2), make_instr(opcode, 2, 0, 1));
+    assert_eq!(bc.get_instr(3), make_instr(Opcode::MOV, 3, 2, 0));
+
+}
+
+#[test]
+fn add_generation() {
+    let pt = LuaParseTree::new("./tests/lua_files/add.lua");
```

Comment:
It would be easier if these test files were inline (since they're such small strings): readers could see the test and its results. You might want to add a `from_str` file to `LuaParseTree` or similar?